lib/bundler/ui/silent.rb

```diff
@@ -13,6 +17,7 @@ def confirm(message, newline = nil)
       end
 
       def warn(message, newline = nil)
+        @warnings |= [message]
```

Comment:
Seems like this will mean that warning message strings will never be garbage collected?